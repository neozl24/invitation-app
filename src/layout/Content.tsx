import { Button } from '../components/Button.tsx'

export const Content = () => {
  return (
    <div className='flex-1 flex flex-col items-center justify-center p-4'>
      <h1 className='text-center text-neutral-500 font-medium text-3xl sm:text-5xl'>
        A better way <br />to enjoy every day.
      </h1>
      <p className='my-8 text-center text-neutral-400 text-lg'>Be the first to know when we launch</p>
      <Button>Request an invite</Button>
    </div>
  )
}