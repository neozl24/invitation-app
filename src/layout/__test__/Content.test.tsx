import { render, screen, fireEvent } from '@testing-library/react'
import { Content } from '../Content'

vi.mock('../../components/features/InvitationModal.tsx', () => ({
  InvitationModal: ({
    onClose,
    onSuccess
  }: {
    onClose: () => void,
    onSuccess: () => void,
  }) => (
    <div>
      <h2>Invitation Modal</h2>
      <button data-testid="request-success" onClick={onSuccess}>
        Success
      </button>
      <button data-testid="close-invitation-modal" onClick={onClose}>
        Close
      </button>
    </div>
  ),
}))

vi.mock('../../components/features/SuccessModal.tsx', () => ({
  SuccessModal: ({ onClose }: { onClose: () => void }) => (
    <div>
      <h2>Success Modal</h2>
      <button data-testid="close-success-modal" onClick={onClose}>
        Close
      </button>
    </div>
  ),
}))

test('renders Content component', () => {
  const { asFragment } = render(<Content />)

  expect(asFragment()).toMatchSnapshot()

  expect(screen.getByText(/A better way/)).toBeInTheDocument()

  const button = screen.getByRole('button', {
    name: 'Request an invite'
  })
  expect(button).toBeInTheDocument()
})

test('opens InvitationModal on button click', () => {
  render(<Content />)

  const button = screen.getByRole('button', {
    name: 'Request an invite'
  })
  fireEvent.click(button)

  expect(screen.getByText('Invitation Modal')).toBeInTheDocument()

  // Trigger onClose callback in mock InvitationModal
  const closeModalButton = screen.getByTestId('close-invitation-modal')
  fireEvent.click(closeModalButton)
  expect(screen.queryByText('Invitation Modal')).not.toBeInTheDocument()
})

test('shows SuccessModal after invitation successfully', () => {
  render(<Content />)

  const button = screen.getByRole('button', {
    name: 'Request an invite'
  })
  fireEvent.click(button)

  // Trigger onSuccess callback in mock InvitationModal
  const successButton = screen.getByTestId('request-success')
  fireEvent.click(successButton)

  // Mock SuccessModal should be rendered
  expect(screen.getByText('Success Modal')).toBeInTheDocument()

  const closeButton = screen.getByTestId('close-success-modal')
  fireEvent.click(closeButton)
  expect(screen.queryByText('Success Modal')).not.toBeInTheDocument()
})
