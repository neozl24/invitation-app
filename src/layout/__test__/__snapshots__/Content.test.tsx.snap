// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Content component 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col items-center justify-center p-4"
  >
    <h1
      class="text-center text-neutral-500 font-medium text-3xl sm:text-5xl"
    >
      A better way 
      <br />
      to enjoy every day.
    </h1>
    <p
      class="my-8 text-center text-neutral-400 text-lg"
    >
      Be the first to know when we launch
    </p>
    <button
      class="
    flex
    items-center
    justify-center
    h-[50px]
    px-4
    border
    text-sm
    font-medium
    outline-none
   
    text-neutral-500
    border-neutral-600
    focus-visible:outline-blue-700
    focus-visible:outline-2
    hover:bg-neutral-100
    active:hover:bg-neutral-50
   mb-8"
    >
      Request an invite
    </button>
  </div>
</DocumentFragment>
`;
